csrc/scheduler/expr_eval_sched.cpp

```diff
@@ -0,0 +1,47 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+
+#include <ir/utils.h>
+#include <scheduler/debug_utils.h>
+#include <scheduler/expr_eval_sched.h>
+#include <scheduler/registry_utils.h>
+
+namespace nvfuser {
+
+ExprEvalScheduler::ExprEvalScheduler(
+    Fusion* fusion,
+    SchedulerRuntimeInfo& runtime_info,
+    HeuristicSummary* data_cache)
+    : SchedulerEntry(heuristicType()) {
+  params_ = std::make_shared<HeuristicParams>("", runtime_info.getIndexType());
```

Comment:
Do you need this line? Also: I would place this in the header.

Reply:
Interesting. I wonder if you could just force `DataType::Int` for the index type always with this scheduler. Note I'm not exactly sure how to do that and it might be more work than this single line.

Reply:
Nevermind, I think this line is fine. You _could_ ignore the runtime_info index type and force int64, but that's orthogonal to whether this line should exist. I think you're good to go.